test/sqrt_test.cpp

```diff
@@ -0,0 +1,84 @@
+//  (C) Copyright Matt Borland 2021.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <cmath>
+#include <limits>
+#include <type_traits>
+#include <boost/math/ccmath/sqrt.hpp>
+#include <boost/core/lightweight_test.hpp>
+
+template <typename Real>
+void test_float_sqrt()
+{
+    using std::abs;
+    
+    constexpr Real tol = 2*std::numeric_limits<Real>::epsilon();
+    
+    constexpr Real test_val = boost::math::ccmath::sqrt(Real(2));
+    constexpr Real dummy = 1;
+    static_assert(test_val > dummy, "Not constexpr");
+
+    Real known_val = std::sqrt(Real(2));
+    BOOST_TEST(abs(test_val - known_val) < tol);
```

Comment:
I would highly recommend using my `math_unit_test.hpp` for this one. The expectation will be that this is half-ulp accurace. Use:

`CHECK_ULP_CLOSE(test_val, known_val, 0)`